contrib/IECoreArnold/src/IECoreArnold/ParameterAlgo.cpp

```diff
@@ -125,9 +125,20 @@ void setParameterInternal( AtNode *node, const char *name, int parameterType, bo
 				}
 				break;
 			case AI_TYPE_ENUM :
-				if( const StringData *data = dataCast<StringData>( name, value ) )
+				// Arnold supports setting enums with either the integer index or the string name
+				if( runTimeCast<const StringData>( value ) )
 				{
-					AiNodeSetStr( node, name, data->readable().c_str() );
+					if( const StringData *data = dataCast<StringData>( name, value ) )
```

Comment:
Isn't this second cast redundant, since internally dataCast is doing the same `runTimeCast` we just did? I think just a single`if( const StringData *data = runTimeCast<const StringData>( value ) )` should be sufficient.